spyder/widgets/ipythonconsole/client.py

```diff
@@ -378,7 +386,25 @@ def restart_kernel(self):
     @Slot(str)
     def kernel_restarted_message(self, msg):
         """Show kernel restarted/died messages."""
-        stderr = codecs.open(self.stderr_file, 'r', encoding='utf-8').read()
+        try:
+            stderr = codecs.open(self.stderr_file, 'r',
+                                 encoding='utf-8').read()
+        except UnicodeDecodeError:
+            # This handling is needed since the stderr file could be encoded
+            # in something different to utf-8. In case of fail at least
+            # we raise an informative exception.
+            # See issue 4191
+            try:
+                encoding = get_coding(open(self.stderr_file, 'rb').read())
+                stderr = codecs.open(self.stderr_file, 'r',
+                                     encoding=encoding).read()
+            except UnicodeDecodeError:
+                try:
+                    stderr = codecs.open(self.stderr_file, 'r',
+                                         encoding='cp437').read()
+                except UnicodeDecodeError:
+                    raise EncodingError("Could not read the stderr "
+                                        "file of the kernel while restarting.")

```

Comment:
and remove this whole block, i.e

```python
        except UnicodeDecodeError:
                try:
                    stderr = codecs.open(self.stderr_file, 'r',
                                         encoding='cp437').read()
                except UnicodeDecodeError:
                    raise EncodingError("Could not read the stderr "
                                        "file of the kernel while restarting.")
```

and change it to

```python
except:
    stderr = None
```

The thing is `cp437` is another particular encoding, and we should have detected it in the previous block. If we're not able to read `self.stderr_file`, then we need to skip this,